tests/kdc/check-cc.in

```diff
@@ -144,36 +144,64 @@ echo "getting tickets (KCM)"; > messages.log
+${kdestroy} -A
+${kinit} --unique foo@${R} || { ec=1 ; eval "${testfailed}"; }
+${kinit} --search --verbose foo@${R} 2>&1 | grep '^Using cache: KCM:[0-9][0-9]*:[0-9]' || { ec=1 ; eval "${testfailed}"; }
+${kdestroy} -A
+${kinit} --default-for-principal foo@${R} || { ec=1 ; eval "${testfailed}"; }
+${kinit} --default-for-principal bar@${R} || { ec=1 ; eval "${testfailed}"; }
+${kinit} bar@${R} || { ec=1 ; eval "${testfailed}"; }
 ${klist} > /dev/null || { ec=1 ; eval "${testfailed}"; }
 ${klist} | grep bar@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
 ${klist} | grep foo@${R} > /dev/null && { ec=1 ; eval "${testfailed}"; }
 ${klist} -l | grep bar@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
 ${klist} -l | grep foo@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
 ${klist} -c KCM: | grep bar@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
 KRB5CCNAME=KCM: ${klist} | grep bar@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
-if [ -n "$BASH_VERSION" ]; then
-    ${klist} -c KCM:${UID} | grep bar@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
-    ${klist} -c KCM:${UID}: | grep bar@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
-    KRB5CCNAME=KCM:${UID} ${klist} | grep bar@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
-    KRB5CCNAME=KCM:${UID}: ${klist} | grep bar@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
+uid=$(id -u)
+if [ -n "$uid" ]; then
+    ${klist} -c KCM:${uid} | grep bar@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
+    KRB5CCNAME=KCM:${uid} ${klist} | grep bar@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
 fi
 ${kdestroy} -A
+${kinit} bar@${R} ${klist} | grep bar@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
+${kdestroy} -A
 ${klist} 2>/dev/null && { ec=1 ; eval "${testfailed}"; }
 ${klist} -l | grep bar@${R} > /dev/null && { ec=1 ; eval "${testfailed}"; }
 ${klist} -l | grep foo@${R} > /dev/null && { ec=1 ; eval "${testfailed}"; }
 ${kinit} bar@${R}
 ${kinit} --default-for-principal foo@${R}
 ${klist} > /dev/null || { ec=1 ; eval "${testfailed}"; }
-${klist} | grep bar@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
-${klist} | grep foo@${R} > /dev/null && { ec=1 ; eval "${testfailed}"; }
+${klist} | grep bar@${R} > /dev/null && { ec=1 ; eval "${testfailed}"; }
+${klist} | grep foo@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
 ${klist} -l | grep bar@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
 ${klist} -l | grep foo@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
-${klist} -c KCM: | grep bar@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
-KRB5CCNAME=KCM: ${klist} | grep bar@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
+${klist} -c KCM: | grep foo@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
+KRB5CCNAME=KCM: ${klist} | grep foo@${R} > /dev/null || { ec=1 ; eval "${testfailed}"; }
 ${kdestroy} -A
+if [ -n "$uid" ]; then
+    ${kinit} -c KCM:${uid}:foobar --no-change-default foo@${R}
```

Comment:
Perhaps `--no-kswitch` or `--no-change-default-ccache` instead of `--no-change-default`.   The latter is very wordy but I fear 'default' will be ambiguous.

Reply:
We've had `--no-change-default` since 2009 (72f134d14).  I will add an alias -- I too don't love `--no-change-default`.

Reply:
I added `--no-kswitch`.